src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+
+    consec_const_steps = 0 # counts how many consecutives steps have been taken
+                           # with same step size
+
+    # The four local error (LE) or truncation error necessary for variable
+    # order calculations are listed below, which correspond to infinity norms of:
+    #      LE(n+1)_{k-1}, LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2}
+    # from Harier pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0,0.0,0.0,0.0
+
+    #-##########################################################################
+    # Variable Step Variable Order (VSVO) Step Integration Loop
+    #-##########################################################################
+    while  tfinal*tdir > t[n]*tdir
+        #--#####################################################################
+        # Attempt a step with current trial stepsize `h`
+        #--#####################################################################
+        successful_step = false
+        while !successful_step
+            #---################################################################
+            # Make an approriate time step with current trial stepsize `h`
+            #---################################################################
+            #Keep step size constant if `h` is close to `prev_h`
+            if abs(h/prev_h)<2 && abs(h/prev_h)>=1
+                consec_const_steps += 1
+                h = prev_h
+            else
+                consec_const_steps = 0
+            end
+
+            #Don't step past `tfinal`
+            if tdir*(t[n] + h) >= tdir*(tfinal)
+                h = tfinal-t[n]
+            end
+
+            #Make the step
+            t_np1 = t[n]+h
+            #---################################################################
+            # Calculating explicit approximatiion to y_np1 with current step h
+            #---################################################################
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,h,t,t_np1,n,min(n,steporder+2))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,t_np1,n,1:min(max(n-1,1),steporder+1))
+            #Note that  1<= min(max(n-1,1),steporder+1) <= n-1 and steporder+1
+
+            # Calculate next value, y_np1, with order = k = steporder + 1. See
+            # equation (5.5) in Hairer et al, Vol 1, where we note that here
+            # y_np1 = y_{k}(t_{n+1})
+
+            y_np1 = y[n]
+            for j= 1:steporder
+              y_np1 += h*g[j]* ϕstar[2,j]
+            end
+            # TODO: figure out why the following does not work
+            # for j= 1:steporder
+            #     for i=1:length(y_np1)
+            #         y_np1[i] += h*g[j]*ϕstar[2,j][i]
+            #     end
+            # end
+
+            # Compute current estimate for derivative
+            dy_np1=F(t_np1,y_np1)
```

Comment:
As `F` is not in-place it will create a new vector anyway.  Thus it is pointless to then copy this into another memory slot.  Thus delete `dy_np1` in the initialization on line 190: https://github.com/JuliaLang/ODE.jl/pull/106/files#diff-43ccd017e404a50f115b60e06e7f1ac5R190